src/bio2bel/sources/intact.py

```diff
@@ -0,0 +1,357 @@
+# -*- coding: utf-8 -*-
+
+"""This script downloads and parses IntAct data and maps the interaction types to BEL."""
+
+from typing import Dict, Iterable, List
+
+import pandas as pd
+from protmapper.uniprot_client import get_mnemonic
+
+import pybel.dsl
+from bio2bel.utils import ensure_path
+from pybel import BELGraph
+
+from ..constants import INTACT_INCREASES_ACTIONS, INTACT_DECREASES_ACTIONS, INTACT_ASSOCIATION_ACTIONS, \
+    INTACT_BINDS_ACTIONS
+
+EVIDENCE = 'From IntAct'
+SEP = '\t'
+
+MODULE_NAME = 'intact'
+URL = 'ftp://ftp.ebi.ac.uk/pub/databases/intact/current/psimitab/intact.zip'
+path = ensure_path(MODULE_NAME, URL)
+sample_path = '/Users/sophiakrix/Downloads/intact_sample.txt'
+
+ID_INTA = '#ID(s) interactor A'
+ID_INTB = 'ID(s) interactor B'
+INTERACTION_TYPES = 'Interaction Type(s)'
+PUBLICATION_ID = 'Publication Identifier(s)'
+DATABASE_INT_A = 'database_intA'
+DATABASE_INT_B = 'database_intB'
+ONLY_ID_INT_A = 'id_intA'
+ONLY_ID_INT_B = 'id_intB'
+UNIPROTKB = 'uniprotkb'
+SOURCE = 'source'
+TARGET = 'target'
+RELATION = 'relation'
+PUBMED_ID = 'pubmed_id'
+columns_mapping = {
+    '#ID(s) interactor A': SOURCE,
+    'ID(s) interactor B': TARGET,
+    'Interaction Type(s)': RELATION,
+    'Publication Identifier(s)': PUBMED_ID,
+}
+
+
+def _load_file(module_name: str = MODULE_NAME, url: str = URL) -> str:
+    """Load the file from the URL and place it into the bio2bel_sophia directory.
+
+    :param module_name: name of module (database)
+    :param url: URL to file from database
+    :return: path of saved database file
+    """
+    return ensure_path(prefix=module_name, url=url)
+
+
+def _get_my_df() -> pd.DataFrame:
+    """Get my dataframe."""
+    path = _load_file()
+    df = pd.read_csv(path, sep=SEP, compression='zip')
+    return df
+
+
+def rename_columns(df: pd.DataFrame, columns_mapping: Dict) -> pd.DataFrame:
+    """Rename the columns in a dataframe according to the specified values in the dictionary.
+
+    :param df: dataframe with columns to be renamed
+    :param columns_mapping: mapping from original column names to new column names
+    :return: renamed dataframe
+    """
+    return df.rename(columns=columns_mapping)
+
+
+def read_intact_file(df: pd.DataFrame) -> pd.DataFrame:
+    """Read in a .txt file from IntAct containing the entire database and return dataframe with columns \
+    for source, target, relation and pubmed_id.
+
+    :param df: intact dataframe to be preprocessed
+    :return: dataframe with IntAct information
+    """
+    print('Dataframe is being created from the file.')
+
+    # take relevant columns for source, target, relation and PubMed ID
+    df = df.loc[:, [SOURCE, TARGET, RELATION, PUBMED_ID]]
+
+    # drop nan value rows for interactor B
+    df = df.loc[df[TARGET] != '-', :]
+
+    return df
+
+
+def split_to_list(unsplitted_list: str, separator: str = '|') -> List:
```

Comment:
typo - 'splitted' should be 'split'. it's same present and past tense

Reply:
English grammar ... :) got it, thanks

Reply:
no problem. When you make commits that address the comments, it's best to write back the commit hash then click "resolve conversation"